### PR TITLE
fix remaining linter warnings

### PR DESCRIPTION
**Pull request by bennyz**

### CODE REVIEW

File: pkg/controller/plan/adapter/ova/builder.go [hunk 1]
```diff
@@ -490,7 +489,7 @@ func getResourceCapacity(capacity int64, units string) (int64, error) {
 	for i := range items {
 		item := strings.TrimSpace(items[i])
 		if i == 0 && len(item) > 0 && item != "byte" {
-			return 0, errors.New(fmt.Sprintf("units '%s' are invalid, only 'byte' is supported", units))
+			return 0, fmt.Errorf(fmt.Sprintf("units '%s' are invalid, only 'byte' is supported", units))
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ova/builder.go:492]:
```suggestion
			return 0, fmt.Errorf("units '%s' are invalid, only 'byte' is supported", units)
```

File: pkg/controller/plan/adapter/ova/builder.go [hunk 2]
```diff
@@ -502,15 +501,15 @@ func getResourceCapacity(capacity int64, units string) (int64, error) {
 		}
 		nums := strings.Split(item, "^")
 		if len(nums) != 2 {
-			return 0, errors.New(fmt.Sprintf("units '%s' are invalid, item is invalid: %s", units, item))
+			return 0, fmt.Errorf(fmt.Sprintf("units '%s' are invalid, item is invalid: %s", units, item))
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ova/builder.go:504]:
```suggestion
			return 0, fmt.Errorf("units '%s' are invalid, item is invalid: %s", units, item)
```

File: pkg/controller/plan/adapter/ova/builder.go [hunk 3]
```diff
@@ -502,15 +501,15 @@ func getResourceCapacity(capacity int64, units string) (int64, error) {
...
 		}
 		base, err := strconv.Atoi(nums[0])
 		if err != nil {
-			return 0, errors.New(fmt.Sprintf("units '%s' are invalid, base component is invalid: %s", units, item))
+			return 0, fmt.Errorf(fmt.Sprintf("units '%s' are invalid, base component is invalid: %s", units, item))
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ova/builder.go:508]:
```suggestion
			return 0, fmt.Errorf("units '%s' are invalid, base component is invalid: %s", units, item)
```

File: pkg/controller/plan/adapter/ova/builder.go [hunk 4]
```diff
@@ -502,15 +501,15 @@ func getResourceCapacity(capacity int64, units string) (int64, error) {
...
 		}
 		pow, err := strconv.Atoi(nums[1])
 		if err != nil {
-			return 0, errors.New(fmt.Sprintf("units '%s' are invalid, pow component is invalid: %s", units, item))
+			return 0, fmt.Errorf(fmt.Sprintf("units '%s' are invalid, pow component is invalid: %s", units, item))
```
Comment (ahadas) [inline pkg/controller/plan/adapter/ova/builder.go:512]:
```suggestion
			return 0, fmt.Errorf("units '%s' are invalid, pow component is invalid: %s", units, item)
```

File: pkg/controller/provider/container/ocp/collection.go [hunk 5]
```diff
@@ -43,7 +43,12 @@ func (r *StorageClass) Reconcile(ctx context.Context) (err error) {
 		err = liberr.Wrap(err)
 		return
 	}
-	defer tx.End()
+	defer func() {
+		err = tx.End()
+		if err != nil {
+			err = liberr.Wrap(err)
+		}
```
Comment (ahadas) [inline pkg/controller/provider/container/ocp/collection.go:50]: is it really recommended to set the error that within a go-routine?...

File: pkg/controller/provider/container/vsphere/collector.go [hunk 6]
```diff
@@ -356,7 +356,15 @@ func (r *Collector) getUpdates(ctx context.Context) error {
 	if err != nil {
 		return liberr.Wrap(err)
 	}
-	defer pc.Destroy(context.Background())
+	defer func() {
+		err = pc.Destroy(context.Background())
```
Comment (ahadas) [inline pkg/controller/provider/container/vsphere/collector.go:360]: same question here - wouldn't it make sense to use a new variable?

File: pkg/lib/inventory/container/ocp/collector.go [hunk 7]
```diff
@@ -184,7 +184,16 @@ func (r *Collector) start(ctx context.Context) (err error) {
 	if err != nil {
 		return
 	}
-	go r.manager.Start(ctx)
+
+	go func() {
+		err = r.manager.Start(ctx)
```
Comment (ahadas) [inline pkg/lib/inventory/container/ocp/collector.go:189]: and here?

File: pkg/lib/inventory/model/model_test.go [hunk 8]
```diff
@@ -1032,14 +1032,14 @@ func TestWatch(t *testing.T) {
 		for i := 0; i < N; i++ {
 			switch action {
 			case Created:
-				created = append(
+				_ = append(
```
Comment (ahadas) [inline pkg/lib/inventory/model/model_test.go:1035]: can't we drop the `_ = ` part?

File: pkg/lib/inventory/model/model_test.go [hunk 9]
```diff
@@ -1032,14 +1032,14 @@ func TestWatch(t *testing.T) {
...
 					created,
 					TestEvent{
 						action: action,
 						model:  &TestObject{ID: i},
 					})
 			case Updated:
-				updated = append(
+				_ = append(
```
Comment (ahadas) [inline pkg/lib/inventory/model/model_test.go:1042]: and here?

**Comment by yaacov** [comment]
@mnecas @bennyz  hi,

With the changing code, this PR is getting more and more out of date without @bennyz in the team to check and update it
I think we should close this PR because of @bennyz capacity ATM, WDYT ?

**Pull request closed without merging**